build-nix

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i bash shell.nix
```

Comment:
This file looks almost identical to `build.stack.nix.sh` -- what's the difference? Do we need both?

Reply:
I see, it is quite similar. But the obvious difference being that this doesn't require stack nor cabal to be installed on your system and is more direct than using a stack.yaml file to specify the nix configuration.

Reply:
I just copied this in my tree for now and would therefore be the second user of this approach.

Reply:
So, to use this script one will first have to somehow build Hadrian. This feels like an easy way to shoot oneself in the foot by using an outdated Hadrian binary.

It may be a good idea to separate the two concerns somehow: building Hadrian and building GHC, but this PR seems to only address the latter, but not the former, unless I'm missing something?

Reply:
(I've never used nix, so excuse my naive comments!)

Reply:
When you call this script, it enters the environment specified in shell.nix, which "provisions" among other things a local build of hadrian.

Reply:
(and it has everything needed for building ghc, too)

Reply:
Thanks!